scripts/helper_file_generator.py

```diff
@@ -20,482 +20,184 @@
-        if category == 'handle':
-            if alias:
-                self.object_type_aliases.append((name,alias))
-            else:
-                self.object_types.append(name)
-        elif (category == 'struct' or category == 'union'):
-            self.structNames.append(name)
-            self.genStruct(typeinfo, name, alias)
-    #
-    # Generate a VkStructureType based on a structure typename
-    def genVkStructureType(self, typename):
-        # Add underscore between lowercase then uppercase
-        value = re.sub('([a-z0-9])([A-Z])', r'\1_\2', typename)
-        # Change to uppercase
-        value = value.upper()
-        # Add STRUCTURE_TYPE_
-        return re.sub('VK_', 'VK_STRUCTURE_TYPE_', value)
-    #
-    # Check if the parameter passed in is a pointer
-    def paramIsPointer(self, param):
-        ispointer = False
-        for elem in param:
-            if ((elem.tag != 'type') and (elem.tail is not None)) and '*' in elem.tail:
-                ispointer = True
-        return ispointer
-    #
-    # Check if the parameter passed in is a static array
-    def paramIsStaticArray(self, param):
-        isstaticarray = 0
-        paramname = param.find('name')
-        if (paramname.tail is not None) and ('[' in paramname.tail):
-            isstaticarray = paramname.tail.count('[')
-        return isstaticarray
-    #
-    # Retrieve the type and name for a parameter
-    def getTypeNameTuple(self, param):
-        type = ''
-        name = ''
-        for elem in param:
-            if elem.tag == 'type':
-                type = noneStr(elem.text)
-            elif elem.tag == 'name':
-                name = noneStr(elem.text)
-        return (type, name)
-    #
-    # Retrieve the value of the len tag
-    def getLen(self, param):
-        result = None
-        len = param.attrib.get('len')
-        if len and len != 'null-terminated':
-            # For string arrays, 'len' can look like 'count,null-terminated', indicating that we
-            # have a null terminated array of strings.  We strip the null-terminated from the
-            # 'len' field and only return the parameter specifying the string count
-            if 'null-terminated' in len:
-                result = len.split(',')[0]
+            return ''.join(str_to_process.title().split('_')[3:])
+
+    def split_handle_by_capital_case(self, str_to_split):
+        name = str_to_split[2:]
+        name = name.replace('NVX', 'Nvx')
+        # Force vendor tags to be title case so we can split by capital letter
+        for tag in self.vk.vendorTags:
+            name = name.replace(tag, tag.title())
+        # Split by capital letters so we can concoct a VK_DEBUG_REPORT_<>_EXT string out of it
+        return re.split('(?<=.)(?=[A-Z])', f'{name}')
+
+
+    def convert_to_VK_OBJECT_TYPE(self, str_to_convert):
+        return '_'.join(['VK', 'OBJECT', 'TYPE'] + self.split_handle_by_capital_case(str_to_convert)).upper()
```

Comment:
What is this join for? why not just go

```suggestion
        return 'VK_OBJECT_TYPE' + self.split_handle_by_capital_case(str_to_convert)).upper()
```

(same for VK_DEBUG_REPORT_OBJECT_TYPE)